### ISSUE TITLE
The Checkbox component is leveraging the `<fieldset>` tag in a semantically incorrect manner

### ISSUE THREAD

**Issue by nicklemmon** [SparkPost/matchbox#198]
This one is going to be a fairly large refactor! The API for the component shouldn't be impacted severely, however.

The `<fieldset>` should be used to group a related series of form controls, as such, a fieldset is invalid if it contains only one child form control. This element is most used when grouping checkboxes or radio buttons.

```html
<fieldset>
  <legend>My Favorite Foods</legend>

  <label for="bananas">
    <input id="bananas" name="bananas" type="checkbox"/>

    Bananas
  </label>

  <label for="apples">
    <input id="apples" name="apples" type="checkbox"/>

    Apples
  </label>
</fieldset>
```

Not only are the inputs appropriately grouped, but now they are appropriately labelled using the `<legend>` element. To break things down cleanly, I'll write up current issues and possible fixes for each issue with this component from a HTML perspective:

## Issues and Solutions

- [ ] The `<fieldset>` should not be repeated for each `<input/>`. __Solution:__ Ensure that the `<Checkbox.Group/>` contains the relevant HTML, and that each `<CheckBox/>` instance does *not* contain a `<fieldset>`
- [ ] The `<CheckBox.Group/>` lacks a `<legend>` - each group of checkboxes or radio buttons needs a label. __Solution:__ Incorporate the `<legend>` and make it a required prop. It can also be rendered as screen reader only, but it does always need to be present
- [ ] The `<label>` should not come after the `<input>` in the document order. __Solution:__ In order to solve this problem visually, simply wrap the `<input/>` in the `<label>` element

This is a big one! We'll need to be carefully re-working this.

**Comment by jonambas** [comment]
Closing - addressed in hibana
